src/devtools/client/inspector/markup/actions/markup.ts

```diff
@@ -74,63 +73,59 @@ export function setupMarkup(store: UIStore, startAppListening: AppStartListening
     actionCreator: paused,
     effect: async (action, listenerApi) => {
       const { condition, dispatch, getState, cancelActiveListeners, extra } = listenerApi;
-      const { ThreadFront, replayClient, protocolClient } = extra;
+      const { ThreadFront, replayClient } = extra;
 
       cancelActiveListeners();
 
+      // Our `state.markup` section has already been cleared out
+      // by the time we get here, from the `pauseRequestedAt` action.
+
       const originalPauseId = await ThreadFront.getCurrentPauseId(replayClient);
 
-      // every time we pause, clear the existing DOM node info
-      dispatch(reset());
+      async function throwIfCanceled() {
+        // Cancel-aware - see if this is still the current pause
+        // We'll have this added to RTK at some point
+        await listenerApi.delay(0);
```

Comment:
Hey @markerikson – as part of looking into FE-1907, I'm seeing this code and I realize that I don't know what it means. It doesn't look like it "throws" despite the name. Could you help me understand what it does?